File: UI/api/repository/models/approach-process.ts
/* tslint:disable */
/* eslint-disable */
import { AnalysisType } from './analysis-type';
import { AtomarUnit } from './atomar-unit';
import { AutomationLevel } from './automation-level';
import { Direction } from './direction';
import { ProcessStrategy } from './process-strategy';
import { Quality } from './quality';
import { QualitySublevel } from './quality-sublevel';
import { Representation } from './representation';
import { Technique } from './technique';
export interface ApproachProcess {
  analysisTypes?: null | Array<AnalysisType>;
  approachProcessId?: number;
  automationLevels?: null | Array<AutomationLevel>;
  directions?: null | Array<Direction>;
  qualities?: null | Array<Quality>;
  qualitySublevels?: null | Array<QualitySublevel>;
  techniques?: null | Array<Technique>;
  processStrategies?: null | Array<ProcessStrategy>;
  atomarUnits?: null | Array<AtomarUnit>;
}

File: ArchitectureRefactoringHelper/Repository/Models/Database/ApproachProcess.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Repository.Models.Database;

[Table(Constants.TableNameApproachProcess)]
public class ApproachProcess
{
    [Key]
    [JsonPropertyName("approachProcessId")]
    public int ApproachProcessId { get; set; }
    [JsonIgnore]
    public RefactoringApproach? RefactoringApproach { get; set; }
    [JsonIgnore]
    public ArchitecturalDesign? ArchitecturalDesign { get; set; }
    
    [JsonPropertyName("qualities")]
    public ICollection<Quality>? Qualities { get; set; }
    [JsonPropertyName("qualitySublevels")]
    public ICollection<QualitySublevel>? QualitySublevels { get; set; }
    [JsonPropertyName("directions")]
    public ICollection<Direction>? Directions { get; set; }
    [JsonPropertyName("automationLevels")]
    public ICollection<AutomationLevel>? AutomationLevels { get; set; }
    [JsonPropertyName("analysisTypes")]
    public ICollection<AnalysisType>? AnalysisTypes { get; set; }
    [JsonPropertyName("techniques")]
    public ICollection<Technique>? Techniques { get; set; }
    [JsonPropertyName("processStrategies")]
    public ICollection<ProcessStrategy>? ProcessStrategies { get; set; }
    [JsonPropertyName("atomarUnits")]
    public ICollection<AtomarUnit>? AtomarUnits { get; set; }
}